### PR TITLE
set up es6 syntax capabilities and linting

### PR DESCRIPTION
**Pull request by AliakseiBychyk**
Set up babel compiler to support es6 syntax, linting, and linter against staged git files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,16 +3,34 @@
...
+    "babel-cli": "^6.26.0",
+    "babel-polyfill": "^6.26.0",
     "mocha": "^2.3.4"
   },
-  "repository" : {
-    "type" : "git",
-    "url" : "https://github.com/it-shark-pro/js-assignments.git"
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/it-shark-pro/js-assignments.git"
```
Comment (maximuk) [inline package.json]: please use `git+ssh` protocol

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "env": {
+    "browser": true,
```
Comment (maximuk) [inline .eslintrc]: `browser` env is redundant

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "es6": true,
+    "node": true
+  },
+  "extends": ["airbnb-base"],
+  "rules": {
+    "max-len": 0,
+    "no-unused-vars": 0,
```
Comment (maximuk) [inline .eslintrc]: what is the reason to disable the rule?

File: .eslintrc (hunk 2 above)
Comment (AliakseiBychyk) [reply .eslintrc]: you're right we execute only in console

File: .eslintrc (hunk 3 above)
Comment (AliakseiBychyk) [reply .eslintrc]: ESLint detects an error while function's parameters not yet exist in implementation.

File: task/01-strings-tasks.js [hunk 4]
```diff
@@ -21,9 +19,10 @@
  *   'aa',''    => 'aa'
  *   '',  'bb'  => 'bb'
  */
-function concatenateStrings(value1, value2) {
```
Comment (EduardTrutsyk) [inline task/01-strings-tasks.js]: I reckon, we can leave normal functions here.

File: task/12-katas-3-tasks.js [hunk 5]
```diff
@@ -65,45 +65,45 @@ function* getPermutations(chars) {
...
  * @example
- *    
+ *
  *     var urlShortener = new UrlShortener();
  *     var shortLink = urlShortener.encode('https://en.wikipedia.org/wiki/URL_shortening');
  *     var original  = urlShortener.decode(shortLink); // => 'https://en.wikipedia.org/wiki/URL_shortening'
- * 
+ *
  */
 function UrlShortener() {
```
Comment (EduardTrutsyk) [inline task/12-katas-3-tasks.js]: maybe better to use `class` syntax in this case?

File: task/01-strings-tasks.js (hunk 4 above)
Comment (AliakseiBychyk) [reply task/01-strings-tasks.js]: There is a question whether it is better to stay with functional declarations or move to functional expressions
I believe that functional expression is more appropriate approach.

File: package.json [hunk 6]
```diff
@@ -3,16 +3,35 @@
...
   "devDependencies": {
+    "babel-cli": "^6.26.0",
+    "babel-polyfill": "^6.26.0",
     "mocha": "^2.3.4"
   },
-  "repository" : {
-    "type" : "git",
-    "url" : "https://github.com/it-shark-pro/js-assignments.git"
+  "repository": {
+    "type": "git+ssh",
```
Comment (maximuk) [inline package.json]: selection_148

File: task/01-strings-tasks.js (hunk 4 above)
Comment (AliakseiBychyk) [reply task/01-strings-tasks.js]: Taking into account the educational nature of the app, I agree that functional declaration is better.

**Pull request merged**